packages/components/src/navigable-container/menu.tsx

```diff
@@ -52,7 +58,11 @@ export function NavigableMenu(
 			stopNavigationEvents
 			onlyBrowserTabstops={ false }
 			role={ role }
-			aria-orientation={ role === 'presentation' ? null : orientation }
+			aria-orientation={
+				role === 'presentation'
+					? undefined
+					: ( orientation as 'vertical' | 'horizontal' )
```

Comment:
Prevents an error from `aria-orientation` not allowing a `'both'` value:

```sh
Type '{ children?: ReactNode; cycle?: boolean | undefined; onNavigate?: ((index: number, focusable: HTMLElement) => void) | undefined; onKeyDown?: ((event: KeyboardEvent) => void) | undefined; ... 256 more ...; eventToOffset: (evt: KeyboardEvent) => 1 | ... 2 more ... | undefined; }' is not assignable to type 'Omit<Pick<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, "key" | keyof HTMLAttributes<HTMLDivElement>>, "as" | ... 4 more ... | "stopNavigationEvents">'.
  Types of property ''aria-orientation'' are incompatible.
    Type '"vertical" | "horizontal" | "both" | undefined' is not assignable to type '"vertical" | "horizontal" | undefined'.
      Type '"both"' is not assignable to type '"vertical" | "horizontal" | undefined'.ts(2322)
```

Reply:
To prevent this `as`, maybe `NavigableMenuProps` could be a discriminated union, forcing `role` to be `'presentation'` whenever `orientation` is `'both'`

But that's a lot of complexity for a small problem.

Reply:
Good question, let me think about this.

Reply:
>…what happens currently when `orientation` is set to `both`?

Nothing, I think 😄 

> If that's an invalid value for the aria-orientation prop, does it mean that it simply gets ignored? 

Yes, I think

It probably depends on whether assistive technologies would recognize 'both'.

But it's hard to see how they would recognize that, as it's not in the [spec](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-orientation) for `aria-orientation`.

<img width="1087" alt="Screenshot 2023-04-06 at 10 38 45 AM" src="https://user-images.githubusercontent.com/4063887/230429019-7dda21c1-5079-4ad7-8a55-14b3993d549b.png">

Reply:
>Based on the current behaviour, we could rewrite the code to keep that same behaviour, while complying to the TS types.

Good idea, what do you think about https://github.com/WordPress/gutenberg/pull/49377/commits/20f9727b5c62778605b92ff8714dd86877ccd4df?

Reply:
Looks good, thank you!